src/big_segment_store.ts

```diff
@@ -0,0 +1,76 @@
+import { BaseRedis } from "./base";
+import { LDRedisOptions } from "./options";
+
+import * as ld from "launchdarkly-node-server-sdk";
+import { RedisClient } from "redis";
+import { promisify } from "util";
+
+export const keyLastUpToDate = "big_segments_synchronized_on";
+export const keyUserInclude = "big_segment_include:";
+export const keyUserExclude = "big_segment_exclude:";
+
+/**
+ * Configures a big segment store backed by a Redis instance.
+ *
+ * "Big segments" are a specific type of user segments. For more information, read the
+ * LaunchDarkly documentation about user segments: https://docs.launchdarkly.com/home/users
+ *
+ * @param options The standard options supported for all LaunchDarkly Redis features, including both
+ *   options for Redis itself and others related to the SDK"s behavior.
+ */
+export function RedisBigSegmentStore(options?: LDRedisOptions):
+    (config: ld.LDOptions) => ld.interfaces.BigSegmentStore {
```

Comment:
The reason that this is a function returning another factory function, rather than just being a constructor, has to do with how the SDK manages the lifecycle of add-on components and passes its state into those components. Application code would not be constructing instances of `BigSegmentStoreImpl` (or `RedisFeatureStoreImpl` in the other file) directly.